test/websocket/utils/protocolClient.js

```diff
@@ -0,0 +1,351 @@
+'use strict';
+
+const net = require('net');
+const crypto = require('crypto');
+
+const TWO_32 = 4294967296;
+
+function parseStatusCode(statusLine) {
+  if (!statusLine) return null;
+  const parts = statusLine.split(' ');
+  const code = parseInt(parts[1], 10);
+  return Number.isFinite(code) ? code : null;
+}
+
+// Minimal low-level client for protocol tests.
+// Events: 'open', 'message'(Buffer), 'frame'(opcode, Buffer), 'ping'(Buffer),
+// 'pong'(Buffer), 'close'(code, reason)
+// Methods: send(data), sendFrame(opcode, payload, { fin=true, mask=true }),
+// sendText, sendBinary, ping, close
+class ProtocolClient {
+  constructor(url) {
```

Comment:
Wery long constructor, maybe we can decompose, for example, extracting collection:
`{ [opcode]: [handler] }`